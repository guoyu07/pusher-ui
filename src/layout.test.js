import React from 'react';
import renderer from 'react-test-renderer';

import { Block, Flex, Inline, InlineBlock } from './index';

describe('Layout', () => {
  const generateTestsFor = (Component, otherTests = () => {}) => {
    describe(Component.displayName, () => {
      const createSnapshot = (props = {}) => {
        expect(
          renderer.create(<Component {...props} />).toJSON(),
        ).toMatchSnapshot();
      };

      it('should render a div with the proper display attribute', () => {
        createSnapshot();
      });

      it('should render children', () => {
        createSnapshot({
          children: 'Hello, world',
        });
      });

      it('should take any css property as props', () => {
        createSnapshot({
          backgroundColor: 'hotpink',
          padding: 20,
        });
      });

      otherTests(createSnapshot);
    });
  };

  generateTestsFor(Block);

  generateTestsFor(Inline);

  generateTestsFor(InlineBlock);

  generateTestsFor(Flex, createSnapshot => {
    it('should have a default prop for flexDirection set to row', () => {
      expect(Flex.defaultProps).toEqual({ flexDirection: 'row' });
    });

    it('should take a gutter prop to determine a gutter between items', () => {
      // Note: it appears that the CSS for selecting the child items is not
      // being applied to the snapshots, so this test doesn't do anything.
      const children = [<Block key="a" />, <Block key="b" />];
      createSnapshot({ gutter: 24, children });

      createSnapshot({ flexDirection: 'column', gutter: 24, children });
    });
  });
});
