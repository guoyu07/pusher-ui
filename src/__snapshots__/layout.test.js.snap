// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout layout__Block should render a div with the proper display attribute 1`] = `
.glamor-0,
[data-glamor-0] {
  display: block;
}

<div
  className="glamor-0"
/>
`;

exports[`Layout layout__Block should render children 1`] = `
.glamor-0,
[data-glamor-0] {
  display: block;
}

<div
  className="glamor-0"
>
  Hello, world
</div>
`;

exports[`Layout layout__Block should take any css property as props 1`] = `
.glamor-0,
[data-glamor-0] {
  display: block;
  background-color: hotpink;
  padding: 20px;
}

<div
  className="glamor-0"
/>
`;

exports[`Layout layout__Flex should render a div with the proper display attribute 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}

<div
  className="glamor-0"
/>
`;

exports[`Layout layout__Flex should render children 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}

<div
  className="glamor-0"
>
  Hello, world
</div>
`;

exports[`Layout layout__Flex should take a gutter prop to determine a gutter between items 1`] = `
.glamor-0,
[data-glamor-0] {
  display: block;
}

.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}

<div
  className="glamor-2"
>
  <div
    className="glamor-0"
  />
  <div
    className="glamor-0"
  />
</div>
`;

exports[`Layout layout__Flex should take a gutter prop to determine a gutter between items 2`] = `
.glamor-0,
[data-glamor-0] {
  display: block;
}

.glamor-2,
[data-glamor-2] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

<div
  className="glamor-2"
>
  <div
    className="glamor-0"
  />
  <div
    className="glamor-0"
  />
</div>
`;

exports[`Layout layout__Flex should take any css property as props 1`] = `
.glamor-0,
[data-glamor-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  background-color: hotpink;
  padding: 20px;
  flex-direction: row;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -webkit-flex-direction: row;
}

<div
  className="glamor-0"
/>
`;

exports[`Layout layout__Inline should render a div with the proper display attribute 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline;
}

<div
  className="glamor-0"
/>
`;

exports[`Layout layout__Inline should render children 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline;
}

<div
  className="glamor-0"
>
  Hello, world
</div>
`;

exports[`Layout layout__Inline should take any css property as props 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline;
  background-color: hotpink;
  padding: 20px;
}

<div
  className="glamor-0"
/>
`;

exports[`Layout layout__InlineBlock should render a div with the proper display attribute 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
}

<div
  className="glamor-0"
/>
`;

exports[`Layout layout__InlineBlock should render children 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
}

<div
  className="glamor-0"
>
  Hello, world
</div>
`;

exports[`Layout layout__InlineBlock should take any css property as props 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  background-color: hotpink;
  padding: 20px;
}

<div
  className="glamor-0"
/>
`;
